nixos/modules/services/networking/bind.nix

```diff
@@ -86,12 +92,19 @@ let
                    ''
                 }
                 allow-query { any; };
+                ${optionalString allowUpdates "allow-update { key rndc-key; };"}
                 ${extraConfig}
               };
             '')
           (attrValues cfg.zones) }
     '';
 
+    copyUpdateableZones = optionalString ((length updateableZones) > 0) ''
```

Comment:
When a zone is removed from the list of update-able zones, then its data file in /var/db/bind will not be removed either, which will result in dead files.